src/main/java/com/aws/greengrass/util/S3SdkClientFactory.java

```diff
@@ -34,16 +46,41 @@
      * @param credentialsProvider credential provider from TES
      */
     @Inject
-    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.NullAssignment"})
     public S3SdkClientFactory(DeviceConfiguration deviceConfiguration, LazyCredentialProvider credentialsProvider) {
         this.credentialsProvider = credentialsProvider;
+        deviceConfiguration.onAnyChange((what, node) -> {
```

Comment:
this is not needed, the only thing which can affect s3 is when the region changes, but even then it doesn't matter

Reply:
The validation used here is deviceConfiguration.validate() which validates all these parameters. If the validate() is false (which it will be in case device is not provisioned at the start) then the client will not be initialized. So need to ensure that when the device configuration becomes valid (which can happen with either of these parameter being set), the client is initialized.

Reply:
Why wouldn't it be initialized? We catch the device validation error in the existing code.